.github/workflows/ci.yml

```diff
@@ -4,6 +4,9 @@ on:
   schedule:
     # At 09:00 on Monday. (see https://crontab.guru)
     - cron: '0 9 * * 1'
+  pull_request:
+    branches:
+      - main
```

Comment:
To remove before merging

Reply:
I'd rather have a few tests at least on PRs, otherwise we might merge broken PRs.

Reply:
> But keeping it would mess up ccache cache.

Cache is not migrated from PRs to `main`, but cache from `main` can be used in PRs, as a security feature.

> We have limited resources from github runners

We can use up to [20 concurrent runners](https://docs.github.com/en/actions/administering-github-actions/usage-limits-billing-and-administration#usage-limits), right now we're not even at 1. I think this mattered a lot more when setting up the builds.

> We should have separate run-test CI job in the future if we want to run any.

This seems like a valid argument -- only test for a given version.

Reply:
I can work on adding a test job, but I need to clean things up a bit more.

Reply:
Some discussions should take place on how to approach it but I also think it isn't a priority right now, considering that we don't have virtually any tests. So I don't think it's worth spending time on right now.

Reply:
Existing job's purpose is to build LLVM and dialect wheel only. We don't have actual dialect feature yet so there's nothing to test yet.

Reply:
I've reverted this in https://github.com/finch-tensor/Finch-mlir/pull/16/commits/2f0e7977ab0d9da21000af24aa3ee29acbfc03f3.